webserver/src/service/pos.rs

```diff
@@ -302,6 +304,51 @@ impl PosService {
         Ok(rewards)
     }
 
+    pub async fn get_rewards_by_delegator_and_validator_and_epoch(
+        &self,
+        delegator: String,
+        validator_address: String,
+        epoch: u64,
+    ) -> Result<Vec<Reward>, PoSError> {
+        // 1) Lookup validator by its address
+        let db_validator = self
+            .pos_repo
+            .find_validator_by_address(validator_address)
+            .await
+            .map_err(PoSError::Database)?;
+
+        // If no validator is found for the given address, handle it as an error
+        // or return empty results
+        let db_validator = match db_validator {
+            Some(val) => val,
+            None => {
+                tracing::error!("No validator found for the given address");
```

Comment:
this `tracing::error!()` will pollute the logs of the webserver for no good reason (it doesn't give any meaningful information about the failure). let's change it to a debug log and include the address of the queried validator.

```suggestion
                tracing::debug!(%validator_address, "No validator found for the given address");
```